### ISSUE TITLE
[BUG] Indexer broken, only synchronizes parent products 

### ISSUE THREAD

**Issue by Swahjak** [Nosto/nosto-magento2#733]
**Magento 2 version(s) used**: 2.4.2
**Extension version(s) affected**: >= https://github.com/Nosto/nosto-magento2/releases/tag/5.2.2

**Description**  

**How to reproduce**  

**Magento 2 mode**
- [x] Developer 
- [x] Production

**Full page cache**
- [x] Enabled 
- [x] Disabled

**Nosto indexer mode**
- [ ] Update On Save  
- [x] Update By Schedule

**Possible Solution**  

I haven't thoroughly checked this, but from the code I don't see how this could work:

https://github.com/Nosto/nosto-magento2/blob/a4ac19c8703bb97b1dedd782807f7a3cdf8e4829/Model/Service/Update/QueueService.php#L171 tries to retrieve the parent product ids, catches a ParentProductDisabledException here https://github.com/Nosto/nosto-magento2/blob/a4ac19c8703bb97b1dedd782807f7a3cdf8e4829/Model/Service/Update/QueueService.php#L172 and then continues here https://github.com/Nosto/nosto-magento2/blob/a4ac19c8703bb97b1dedd782807f7a3cdf8e4829/Model/Service/Update/QueueService.php#L174. However form what I can tell https://github.com/Nosto/nosto-magento2/blob/a4ac19c8703bb97b1dedd782807f7a3cdf8e4829/Model/Product/Repository.php#L193 always throws when the product type is `simple` and it doesn't have any parents.

**Additional context**

**Comment by wouter-toppy** [comment]
https://github.com/Nosto/nosto-magento2/blob/master/Model/Service/Update/QueueService.php#L164

```php
    private function toParentProductIds(ProductCollection $collection)
    {
        $productIds = [];
        /** @var ProductInterface $product */
        foreach ($collection->getItems() as $product) {
            try {
                /** @phan-suppress-next-line PhanTypeMismatchArgument */
                $parents = $this->nostoProductRepository->resolveParentProductIds($product);

                if (!empty($parents)) {
                    foreach ($parents as $id) {
                        $productIds[] = $id;
                    }
                }
            } catch (ParentProductDisabledException $e) {
                $this->getLogger()->debug($e->getMessage());
                $productIds[] = $product->getId();
            }
        }
        return array_unique($productIds);
    }
```

**Comment by olsi-qose** [comment]
Thank you @Swahjak ! We will start looking into this issue soon.

**Comment by olsi-qose** [comment]
This issue is now fixed on the latest release `5.2.8`.
